llama-index-integrations/embeddings/llama-index-embeddings-yandexgpt/llama_index/embeddings/yandexgpt/util.py

```diff
@@ -0,0 +1,74 @@
+import json
+from typing import Optional
+
+
+class YException(Exception):
+    """Custom exception for YAuth related errors."""
+
+
+class YAuth:
+    """Class to handle authentication using folder ID and API key."""
+
+    def __init__(self, folder_id, api_key=None, iam_token=None) -> None:
```

Comment:
Similar comment here, nothing is typed

Reply:
Hi, I have updated my pull request according to your remarks. Could you please review it again at your earliest convenience?
Thank you for your time and feedback!